`issue view` tests fails after 2021-01-23

### Describe the bug

The current implementation of `issue view` tests doesn't mock the current time and makes test fail after `2021-01-23`.

### Steps to reproduce the behavior

1. Clone `cli/cli`
2. Type `go test ./pkg/cmd/issue/view`
3. See 
```
FAIL: TestIssueView_tty_Preview (0.00s)
    --- FAIL: TestIssueView_tty_Preview/Open_issue_with_empty_body (0.00s)
        view_test.go:267: output did not match regexp /Open.*marseilles opened about 9 years ago.*9 comments/
            > output
            ix of coins
            Open • marseilles opened about 10 years ago • 9 comments
            Labels: tarot
```
These tests started to fail 2 days ago because the dates in the test fixtures point to `2011-01-26` https://github.com/cli/cli/blob/6e2c1b33b0a046a7290b645e87774752b3a0d45e/pkg/cmd/issue/view/fixtures/issueView_previewWithMetadata.json#L10, which makes https://github.com/cli/cli/blob/6e2c1b33b0a046a7290b645e87774752b3a0d45e/pkg/cmd/issue/view/view.go#L142-L145

output:
`Open • marseilles opened about 10 years ago • 9 comments`
instead of:
`Open • marseilles opened about 9 years ago • 9 comments`.

See: https://github.com/cli/cli/pull/2839/checks?check_run_id=1758707814#step:5:55

--- a/pkg/cmd/issue/view/view.go
+++ b/pkg/cmd/issue/view/view.go
@@ -9,7 +9,6 @@ import (
 
 	"github.com/MakeNowJust/heredoc"
 	"github.com/cli/cli/api"
-	"github.com/cli/cli/internal/config"
 	"github.com/cli/cli/internal/ghrepo"
 	"github.com/cli/cli/pkg/cmd/issue/shared"
 	issueShared "github.com/cli/cli/pkg/cmd/issue/shared"
@@ -23,20 +22,21 @@ import (
 
 type ViewOptions struct {
 	HttpClient func() (*http.Client, error)
-	Config     func() (config.Config, error)
 	IO         *iostreams.IOStreams
 	BaseRepo   func() (ghrepo.Interface, error)
 
 	SelectorArg string
 	WebMode     bool
 	Comments    bool
+
+	Now func() time.Time
 }
 
 func NewCmdView(f *cmdutil.Factory, runF func(*ViewOptions) error) *cobra.Command {
 	opts := &ViewOptions{
 		IO:         f.IOStreams,
 		HttpClient: f.HttpClient,
-		Config:     f.Config,
+		Now:        time.Now,
 	}
 
 	cmd := &cobra.Command{
@@ -141,7 +141,7 @@ func printRawIssuePreview(out io.Writer, issue *api.Issue) error {
 
 func printHumanIssuePreview(opts *ViewOptions, issue *api.Issue) error {
 	out := opts.IO.Out
-	now := time.Now()
+	now := opts.Now()
 	ago := now.Sub(issue.CreatedAt)
 	cs := opts.IO.ColorScheme()
 

--- a/pkg/cmd/issue/view/view_test.go
+++ b/pkg/cmd/issue/view/view_test.go
@@ -7,6 +7,7 @@ import (
 	"net/http"
 	"os/exec"
 	"testing"
+	"time"
 
 	"github.com/cli/cli/internal/config"
 	"github.com/cli/cli/internal/ghrepo"
@@ -251,20 +252,38 @@ func TestIssueView_tty_Preview(t *testing.T) {
 	}
 	for name, tc := range tests {
 		t.Run(name, func(t *testing.T) {
-			http := &httpmock.Registry{}
-			defer http.Verify(t)
+			io, _, stdout, stderr := iostreams.Test()
+			io.SetStdoutTTY(true)
+			io.SetStdinTTY(true)
+			io.SetStderrTTY(true)
 
-			http.Register(httpmock.GraphQL(`query IssueByNumber\b`), httpmock.FileResponse(tc.fixture))
+			httpReg := &httpmock.Registry{}
+			defer httpReg.Verify(t)
 
-			output, err := runCommand(http, true, "123")
-			if err != nil {
-				t.Errorf("error running `issue view`: %v", err)
+			httpReg.Register(httpmock.GraphQL(`query IssueByNumber\b`), httpmock.FileResponse(tc.fixture))
+
+			opts := ViewOptions{
+				IO: io,
+				Now: func() time.Time {
+					t, _ := time.Parse(time.RFC822, "03 Nov 20 15:04 UTC")
+					return t
+				},
+				HttpClient: func() (*http.Client, error) {
+					return &http.Client{Transport: httpReg}, nil
+				},
+				BaseRepo: func() (ghrepo.Interface, error) {
+					return ghrepo.New("OWNER", "REPO"), nil
+				},
+				SelectorArg: "123",
 			}
 
-			assert.Equal(t, "", output.Stderr())
+			err := viewRun(&opts)
+			assert.NoError(t, err)
+
+			assert.Equal(t, "", stderr.String())
 
 			//nolint:staticcheck // prefer exact matchers over ExpectLines
-			test.ExpectLines(t, output.String(), tc.expectedOutputs...)
+			test.ExpectLines(t, stdout.String(), tc.expectedOutputs...)
 		})
 	}
 }

